### PR TITLE
Kafka group id, create producer & Dynamo check

### PR DESCRIPTION
**Pull request by erohmensing**
**Proposed changes**:
- Apply @rgstephens changes from https://github.com/RasaHQ/rasa/pull/6521
- Kafka Producer, add group id & client id
- Kafka Producer save/reuse connection
- patch master fix https://github.com/RasaHQ/rasa/pull/6233 

**Status (please check what you already did)**:
- [ ] added some tests for the functionality
- [ ] updated the documentation
- [ ] updated the changelog (please check [changelog](https://github.com/RasaHQ/rasa/tree/master/changelog) for instructions)
- [x] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa#code-style) for instructions)

### CODE REVIEW

**Comment by erohmensing** [comment]
@rgstephens I rebased the PR so that it can hopefully get a review today, but I'll leave you to handle any review changes 👍

**Comment by rgstephens** [comment]
Thanks @erohmensing for cleaning that up. I fixed a log message issue and the checks are running fine except for the 800+ DeepSource issues.

**Comment by erohmensing** [comment]
Sure thing, I've requested sasha to review though because I'm not as familiar with event brokers. One thing from me though, we need to make sure to add the options to the docs -- we just ran into another user looking for this and searching for it in the docs

**Review by ricwo — COMMENTED** [review]
Are there any changelog entries for the Kafka and Dynamo changes we need to port as well?

File: rasa/core/brokers/kafka.py [hunk 1]
```diff
@@ -71,9 +98,12 @@ def _create_producer(self) -> None:
                 ssl_keyfile=self.ssl_keyfile,
                 ssl_check_hostname=False,
                 security_protocol=self.security_protocol,
+                client_id=self.client_id,
+                group_id=self.group_id,
             )
```
[the remark this answers is not in the document]
Comment (rgstephens) [reply rasa/core/brokers/kafka.py:103]: Adding `PLAINTEXT` and `SASL_SSL`.

File: rasa/core/brokers/kafka.py [hunk 2]
```diff
@@ -46,9 +50,30 @@ def from_endpoint_config(cls, broker_config) -> Optional["KafkaEventBroker"]:
...
+                logger.debug("Failed to connect kafka")
+                return
+        try:
+            self._publish(event)
+        except Exception as e:
+            logger.error(
+                f"Could not publish message to kafka host '{self.host}'."
+                f"Failed with error: {e}"
+            )
+            connected = self.producer.bootstrap_connected()
```
[the remark this answers is not in the document]
Comment (rgstephens) [reply rasa/core/brokers/kafka.py:68]: Adding retries as described in comment below

File: rasa/core/brokers/kafka.py [hunk 3]
```diff
@@ -46,9 +50,30 @@ def from_endpoint_config(cls, broker_config) -> Optional["KafkaEventBroker"]:
...
+            if not connected:
```
[the remark this answers is not in the document]
Comment (rgstephens) [reply rasa/core/brokers/kafka.py:69]: Good idea, adding this...

File: rasa/core/brokers/kafka.py [hunk 4]
```diff
@@ -61,6 +86,8 @@ def _create_producer(self) -> None:
                 sasl_plain_password=self.sasl_password,
                 sasl_mechanism="PLAIN",
                 security_protocol=self.security_protocol,
+                client_id=self.client_id,
+                group_id=self.group_id,
             )
         elif self.security_protocol == "SSL":
```
[the remark this answers is not in the document]
Comment (rgstephens) [reply rasa/core/brokers/kafka.py:92]: Adding...

**Review by b-quachtran — APPROVED** [review]

**Pull request merged**